src/ts/ssh-tcp/services/portForwardingService.ts

```diff
@@ -530,26 +552,30 @@ export class PortForwardingService extends SshService {
 		await super.onSessionRequest(args, cancellation);
 
 		let response: SshMessage | undefined;
+		let localPort: number | null = null;
 		if (args.isAuthorized) {
 			if (request.requestType === PortForwardingService.portForwardRequestType) {
-				let localForwardedPort: number | null;
 				try {
-					localForwardedPort = await this.startForwarding(
+					localPort = await this.startForwarding(
 						localIPAddress,
-						localPort,
+						portForwardRequest.port,
 						cancellation,
 					);
 				} catch (e) {
 					// The error is already traced.
-					localForwardedPort = null;
 				}
-				if (localForwardedPort !== null) {
+				if (localPort !== null) {
+					// The chosen local port may be different from the requested port. Use the
+					// requested port in the response, unless the request was for a random port.
```

Comment:
What does it mean, `request for a random port`? Like  SSH server asks the client, hey, let's forward some random port? What's the user scenario?

Reply:
Yes. The user scenario is they need to open a separate communication channel between client and server and they don't care what the port number will be - though they will find out what the number is when it's chosen. We don't use that for tunnels currently, but it's part of the protocol.